source/includes/_feature-extending.md

```diff
@@ -0,0 +1,39 @@
+## Merging and Joining Datasets
+
+```http
+POST /datasets/{dataset_id}/variables/ HTTP/1.1
+Content-Type: application/json
+
+{
+    "function": "adapt",
+    "args": [
+        {
+            "dataset": "/datasets/{other_id}/",
+            "filter": {},
+            "select": {
+                "function": "identify",
+                "args": [
+                    {
+                        "ids": [...]
+                    }
+                ]
+            }
+        },
+        {
+            "variable": "/datasets/{other_id}/variables/{right_key_id}/"
+        },
+        {
+            "variable": "/datasets/{dataset_id}/variables/{left_key_id}/"
+        },
+        ... // Other future args
+    ]
+}
+
```

Comment:
This is the expression currently coded in the API. The `select` call is munged in by the controller just like deselect is also implicit for removing private variables. Also both are constructed differently https://github.com/Crunch-io/backend-http/blob/d77cc62182d165946180172be0d44c5e5b2156d5/cr/server/api/controllers/variables/catalog.py#L587

```
{
            "function": "adapt",
            "args": [{
                "dataset": "/datasets/{other_id}/",
            }, {
                "variable": "/datasets/{other_id}/variables/{right_key_id}/"
            }, {
                "variable": "/datasets/{dataset_id}/variables/{left_key_id}/"
            }, {
                'variables': [
                    // If present, will only extend the selected variables URLs
                    "/datasets/{other_id}/variables/{variable1_id}/",
                    "/datasets/{other_id}/variables/{variable2_id}/",
                    "/datasets/{other_id}/variables/{variable3_id}/"
                ]
            }]
        }
```